fast_calvo_trainer.py

```diff
@@ -0,0 +1,109 @@
+                'type': 'integer',
+                'minimum': 1,
+                'default': 10
+            },
+            'Patch height': {
+                'type': 'integer',
+                'minimum': 64,
+                'default': 256
+            },
+            'Patch width': {
+                'type': 'integer',
+                'minimum': 64,
+                'default': 256
+            }
+        }
+    }
+
+    input_port_types = (
+        {'name': 'Image', 'minimum': 1, 'maximum': 1, 'resource_types': ['image/rgb+png','image/rgb+jpg']},
+        {'name': 'rgba PNG - Background layer', 'minimum': 1, 'maximum': 1, 'resource_types': ['image/rgba+png']},
+        {'name': 'rgba PNG - Music symbol layer', 'minimum': 1, 'maximum': 1, 'resource_types': ['image/rgba+png']},
+        {'name': 'rgba PNG - Staff lines layer', 'minimum': 1, 'maximum': 1, 'resource_types': ['image/rgba+png']},
+        {'name': 'rgba PNG - Text', 'minimum': 1, 'maximum': 1, 'resource_types': ['image/rgba+png']},
+        {'name': 'rgba PNG - Selected regions', 'minimum': 1, 'maximum': 1, 'resource_types': ['image/rgba+png']}
+    )
+
+    output_port_types = (
+        {'name': 'Background Model', 'minimum': 1, 'maximum': 1, 'resource_types': ['keras/model+hdf5']},
+        {'name': 'Music Symbol Model', 'minimum': 1, 'maximum': 1, 'resource_types': ['keras/model+hdf5']},
+        {'name': 'Staff Lines Model', 'minimum': 1, 'maximum': 1, 'resource_types': ['keras/model+hdf5']},
+        {'name': 'Text Model', 'minimum': 1, 'maximum': 1, 'resource_types': ['keras/model+hdf5']},
+    )
+
+
+    def run_my_task(self, inputs, settings, outputs):
+        # Ports
+        input_image = cv2.imread(inputs['Image'][0]['resource_path'], True) # 3-channel
+        background = cv2.imread(inputs['rgba PNG - Background layer'][0]['resource_path'], cv2.IMREAD_UNCHANGED) # 4-channel
+        notes = cv2.imread(inputs['rgba PNG - Music symbol layer'][0]['resource_path'], cv2.IMREAD_UNCHANGED) # 4-channel
+        lines = cv2.imread(inputs['rgba PNG - Staff lines layer'][0]['resource_path'], cv2.IMREAD_UNCHANGED) # 4-channel
+        text = cv2.imread(inputs['rgba PNG - Text'][0]['resource_path'], cv2.IMREAD_UNCHANGED) # 4-channel
+        regions = cv2.imread(inputs['rgba PNG - Selected regions'][0]['resource_path'], cv2.IMREAD_UNCHANGED) # 4-channel
+
+        # Create categorical ground-truth
+        gt = {}
+        regions_mask = (regions[:, :, 3] == 255)
+
+        notes_mask = (notes[:, :, 3] == 255)
+        gt['symbols'] = np.logical_and(notes_mask, regions_mask) # restrict layer to only the notes in the selected regions
+
+        lines_mask = (lines[:, :, 3] == 255)
+        gt['staff'] = np.logical_and(lines_mask, regions_mask) # restrict layer to only the staff lines in the selected regions
+
+        text_mask = (text[:, :, 3] == 255)
+        gt['text'] np.logical_and(text_mask, regions_mask) # restrict layer to only the text in the selected regions
```

Comment:
This is wrong. 
`gt['text'] =  np.logical_and(text_mask, regions_mask)`

Reply:
Why is that? Only for the text layer?

Reply:
No, the `=` is missing